thriftbp/client_pool.go

```diff
@@ -62,9 +62,29 @@ type ClientPoolConfig struct {
 	// "${host}:${port}"
 	Addr string `yaml:"addr"`
 
-	// InitialConnections is the inital number of thrift connections created by
-	// the client pool.
-	InitialConnections int `yaml:"initialConnections"`
+	// InitialConnections is the desired inital number of thrift connections
+	// created by the client pool.
+	//
+	// If an error occurred when we try to establish the initial N connections for
+	// the pool, we log the error with InitialConnectionsFallbackLogger,
+	// then return the pool with the <N connections we already established.
+	//
+	// If the fallback attempt succeeded, we log the initial error with
+	// InitialConnectionsFallbackLogger, and returns nil error.
+	// If the fallback attempt also failed, we return both errors.
+	//
+	// If <N initial connections is not acceptable, you can set
+	// InitialConnectionsNoFallback to true. In that case we return an error if
+	// any error occurred while trying to establish the initial connections.
+	// But please note that enabling this can cause cascading failures during an
+	// outage, as when the upstream is in an outage, you won't be able to do any
+	// deploys because new pods will fail creating the client pool.
+	InitialConnections               int         `yaml:"initialConnections"`
+	InitialConnectionsFallbackLogger log.Wrapper `yaml:"initialConnectionsFallbackLogger"`
+	InitialConnectionsNoFallback     bool        `yaml:"initialConnectionsNoFallback"`
```

Comment:
Maybe `RequireInitialConnections int` to set how many connections the pool must have in order to be considered successfully constructed?  (the default being zero, i.e. don't fail if you can't establish any connections)